src/clients/HubPoolClient.ts

```diff
@@ -768,11 +768,21 @@ export class HubPoolClient extends BaseAbstractClient {
       searchConfig: eventSearchConfigs.map(({ eventName, searchConfig }) => ({ eventName, searchConfig })),
     });
     const timerStart = Date.now();
-    const [currentTime, pendingRootBundleProposal, ...events] = await Promise.all([
-      this.hubPool.getCurrentTime({ blockTag: searchConfig.toBlock }),
-      this.hubPool.rootBundleProposal({ blockTag: searchConfig.toBlock }),
-      ...eventSearchConfigs.map((config) => paginatedEventQuery(this.hubPool, config.filter, config.searchConfig)),
+
+    const { hubPool } = this;
+    const multicallFunctions = ["getCurrentTime", "rootBundleProposal"];
+    const [multicallOutput, ...events] = await Promise.all([
+      hubPool.callStatic.multicall(
+        multicallFunctions.map((f) => hubPool.interface.encodeFunctionData(f)),
+        { blockTag: searchConfig.toBlock }
+      ),
+      ...eventSearchConfigs.map((config) => paginatedEventQuery(hubPool, config.filter, config.searchConfig)),
     ]);
+    const [currentTime, pendingRootBundleProposal] = multicallFunctions.map((fn, idx) => {
+      const output = hubPool.interface.decodeFunctionResult(fn, multicallOutput[idx]);
+      return BigNumber.isBigNumber(output) ? output.toNumber() : spreadEvent(output);
+    });
```

Comment:
Because we have a small, fixed number of functions that we want to multicall here, could we manually write out the operations? 

```suggestion
    const { hubPool } = this;
    const multicallFunctions = ["getCurrentTime", "rootBundleProposal"];
    const [[currentTimeMulticall, rootBundleMulticall], ...events] = await Promise.all([
      hubPool.callStatic.multicall(
        [
            hubPool.interface.encodeFunctionData("getCurrentTime"),
            hubPool.interface.encodeFunctionData("rootBundleProposal")
        ],
        { blockTag: searchConfig.toBlock }
      ),
      ...eventSearchConfigs.map((config) => paginatedEventQuery(hubPool, config.filter, config.searchConfig)),
    ]);
    const [currentTime, pendingRootBundleProposal] = [
        hubPool.interface.decodeFunctionResult("getCurrentTime", currentTimeMulticall).toNumber(),
        spreadEvent(hubPool.interface.decodeFunctionResult("rootBundleMulticall", rootBundleMulticall))
    ]
```

Reply:
This does make the code more readable, too.

Reply:
I avoided that construction because it requires the ordering of function names to be correct in 3 separate places, two of which are as string literals. The advantage of specifying it up front in the `multicallFunctions` array is that the subsequent decoding is guaranteed to occur in the correct order, so there's only 2 places that need to be aligned (the ordering of `multicallFunctions` and the resulting dereferencing at line 781), rather than 3.